scipy/sparse/_sputils.py

```diff
@@ -362,8 +391,14 @@ def check_shape(args, current_shape=None, *, allow_1d=False) -> tuple[int, ...]:
         else:
             raise ValueError('can only specify one unknown dimension')
 
-    if len(new_shape) != 2 and not (allow_1d and len(new_shape) == 1):
-        raise ValueError('matrix shape must be two-dimensional')
+    if len(new_shape) < 3:
+            if len(new_shape) != 2 and not (allow_1d and len(new_shape) == 1):
+                raise ValueError('matrix shape must be two-dimensional')
+    elif len(new_shape) >= 3:
```

Comment:
Some duplication here too, I suggest:
- define `ndim_new = len(new_shape)` and use `ndim_new` in the checks
- try to unify the checks that yield an identical error message (`'matrix shape must be two-dimensional'` in this case)

Reply:
Addressed at 8ec8bd0.